packages/perseus-editor/src/components/locked-circle-settings.tsx

```diff
@@ -0,0 +1,214 @@
+import {components, lockedCircleFillStyles} from "@khanacademy/perseus";
+import {View, useUniqueIdWithMock} from "@khanacademy/wonder-blocks-core";
+import {OptionItem, SingleSelect} from "@khanacademy/wonder-blocks-dropdown";
+import {TextField} from "@khanacademy/wonder-blocks-form";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import {spacing} from "@khanacademy/wonder-blocks-tokens";
+import {LabelMedium, LabelLarge} from "@khanacademy/wonder-blocks-typography";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+
+import CircleSwatch from "./circle-swatch";
+import ColorSelect from "./color-select";
+import CoordinatePairInput from "./coordinate-pair-input";
+import LockedFigureSettingsAccordion from "./locked-figure-settings-accordion";
+import LockedFigureSettingsActions from "./locked-figure-settings-actions";
+
+import type {AccordionProps} from "./locked-figure-settings";
+import type {
+    LockedCircleFillType,
+    LockedCircleType,
+    LockedFigureColor,
+} from "@khanacademy/perseus";
+
+const {InfoTip} = components;
+
+export type Props = AccordionProps &
+    LockedCircleType & {
+        /**
+         * Called when the delete button is pressed.
+         */
+        onRemove: () => void;
+        /**
+         * Called when the props (coords, color, etc.) are updated.
+         */
+        onChangeProps: (newProps: Partial<LockedCircleType>) => void;
+    };
+
+const LockedCircleSettings = (props: Props) => {
+    const {
+        center,
+        radius,
+        color,
+        fillStyle,
+        strokeStyle,
+        expanded,
+        onToggle,
+        onChangeProps,
+        onRemove,
+    } = props;
+
+    const [radiusInput, setRadiusInput] = React.useState(radius.toString());
+
+    const ids = useUniqueIdWithMock();
+    const radiusInputId = ids.get("radius-input");
+    const strokeSelectId = ids.get("stroke-style-select");
+    const fillSelectId = ids.get("fill-style-select");
+
+    function handleRadiusChange(newValue) {
+        // Update the local state (update the input field value).
+        setRadiusInput(newValue);
+
+        // Personal preference: Stop the previous graph from disappearing
+        // before the user finishes typing a new valid number.
+        //
+        // If the new value is not a number (e.g. "." while starting to type
+        // a decimal value), don't update the props.
+        // If it's empty (e.g. erased the previous value before starting to
+        // type a new value), keep the props the same value instead of
+        // setting to 0.
+        // Without this check, the input would assume a 0 value and make the
+        // figure disappear before the user finishes typing the new value.
+        if (isNaN(+newValue) || newValue === "") {
+            return;
+        }
+
+        // Update the props (update the graph).
+        onChangeProps({radius: +newValue});
```

Comment:
Curious: Why is a plus sign (+) preceding `newValue` in some places, but not others (i.e. line 81)?